data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/AddEntryProcessorConfig.java

```diff
@@ -43,17 +46,24 @@ public boolean getOverwriteIfKeyExists() {
             return overwriteIfKeyExists;
         }
 
+        public String getAddWhen() { return addWhen; }
```

Comment:
Should we allow another `add_when` on the same level as `entries`, which has the benefit of combining multiple entries to add under one condition? Same for other processors with `entries`.

Reply:
We can but I think in the future we will have a global processor "when" option which will be used in that place, so I don't think it's necessary at this time.